### PR TITLE
Solve major corruption issues

### PR DESCRIPTION
**Pull request by tcoupland**
The first two commits are smaller code shuffles, the meat is in the last one. This seems to resolve the obvious problems with our events. 

There is still the chance that actual data has been dropped during the byte munging processes, that has been reduce by the skip-utf-8-encoded-chars fn, but is still possible. The final check will come when we perform the replay, as these problems can only be found with eyeballs at this time.

Customised nippy decoder 
The modified nippy encoder takes care of three things:

1) shorts encoded as multi utf-8 char bytes, see read-md-count
2) corrupted file-size bytes, all file sizes are removed from events,
to be patched up next.
3) Exceptions in events. These are discarded.

### CODE REVIEW

File: src/kixi/event_fixer.clj [hunk 1]
```diff
@@ -184,8 +156,20 @@
...
-   (map file->error-report)
+   (comp (map file->error-report)
+         (filter (fn [report]
+                   (not (zero? (:error (first (vals report))))))))
    report-errors
    (rest
     (file-seq
      (io/file local-new-format-base-dir)))))
+
+(defn peak-files
```
Comment (acron0) [inline src/kixi/event_fixer.clj:167]: The way you explain this in demo made me think this should be `peek`?

File: src/kixi/file_size.clj [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  [file-name]
+  (with-open [in (io/input-stream file-name)]
+    (json/parse-stream (io/reader in) true)))
+
+(defn parse-int
+  [item]
+  (if-let [s ((comp :N :kixi.datastore.metadatastore_size-bytes) item)]
+    (Integer/parseInt s)
+    (do
+      (prn item)
```
Comment (acron0) [inline src/kixi/file_size.clj:41]: Assume this a debug `prn`

File: src/kixi/file_size.clj [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    (->> put-req
+         :PutRequest
+         :Item)))
+
+(defn bundle?
+  [item]
+  (= "bundle"
+     (:S (:kixi.datastore.metadatastore_type item))))
+
+(comment
```
Comment (acron0) [inline src/kixi/file_size.clj:56]: Remove this

File: src/kixi/file_size.clj [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+         (map decode-file)
+         (mapcat extract-items)
+         (remove bundle?)
+         (map (juxt (comp :S :kixi.datastore.metadatastore_id)
+                    parse-int)))
+        (:object-summaries (s3/list-objects credentials (merge {:bucket-name dynamo-backup-bucket
+                                                                :prefix dynamo-backup-prefix
+                                                                :max-keys 20})))))
+
+(def file-id->size
```
Comment (acron0) [inline src/kixi/file_size.clj:69]: Yikes, can we get some information about this var?

File: src/kixi/nippy/tools.clj [hunk 5]
```diff
@@ -0,0 +1,48 @@
+(ns taoensso.nippy.tools
```
Comment (acron0) [inline src/kixi/nippy/tools.clj:1]: Does this ns want to change? If not, why not?

File: src/kixi/uploader.clj [hunk 6]
```diff
@@ -0,0 +1,53 @@
+(ns kixi.uploader
+  (:require [amazonica.aws.s3 :as s3]
+            [kixi.maws :refer [witan-admin-prod-creds]]
+            [clojure.java.io :as io]
+            [clojure.string :as string])
+  (:import [java.io ByteArrayInputStream File InputStream]))
+
+(def mfa "442962")
```
Comment (acron0) [inline src/kixi/uploader.clj:8]: It feels like an env var override might be good here? Obv this mfa isn't going to work

File: src/kixi/event_fixer.clj (hunk 1 above)
Comment (tcoupland) [reply src/kixi/event_fixer.clj:167]: lol

**Comment by acron0** [comment]
Lots of `(comment...` blocks, some of which look like they've come in from code imports. For any that aren't it'd be good to either remove them or get some justification as to why they're worth keeping, what they show etc.

File: src/kixi/file_size.clj (hunk 4 above)
Comment (tcoupland) [reply src/kixi/file_size.clj:69]: the clue is in the name...

File: src/kixi/file_size.clj (hunk 3 above)
Comment (tcoupland) [reply src/kixi/file_size.clj:56]: nope, that's the bit that creates the file-id->size map.

File: src/kixi/nippy/tools.clj (hunk 5 above)
Comment (tcoupland) [reply src/kixi/nippy/tools.clj:1]: Not sure this is actually used by anything, came over as part of the nippy copy

File: src/kixi/file_size.clj (hunk 3 above)
Comment (acron0) [reply src/kixi/file_size.clj:56]: Then please can you just include a comment which indicates that?

File: src/kixi/file_size.clj (hunk 2 above)
Comment (tcoupland) [reply src/kixi/file_size.clj:41]: This is an error report. When the thing runs it should be silent, any output is bad.

File: src/kixi/uploader.clj (hunk 6 above)
Comment (tcoupland) [reply src/kixi/uploader.clj:8]: It should be an arg to the upload fn really.

**Comment by tcoupland** [comment]
Chatted out of band about this. Think it's ok to merge this.

This code represents a utility, one that we hope not to need again. If we do need it it's not so bad that it wouldn't be of some help, but the problem will be novel, so polishing this up to normal standards doesn't feel warrented.

**Pull request merged**